from_mwt_ds/DataScience/tests/test_estimate.py

```diff
@@ -0,0 +1,46 @@
+import unittest, pandas as pd
+from estimate import estimate_bucket
+# df needs to have probability (p), reward (r) and probability of policy to be estimated for counterfactual
+
+# ips = \sum(r * p_est / p) / \sum(1)
+# snips = \sum(r * p_est / p) / \sum(p_est / p)
+
+class TestEstimate(unittest.TestCase):
+    # def test_estimate_bucket(self):
+        # df = pd.DataFrame({'p': [0.5], 'r': [1], 'policy_1': [0.5]})
+        # config = {
+        #     'policies':{'policy_1':['ips','snips']}
+        # }
+
+        # self.assertEqual(
+        #     {'policy_1_ips': [1], 'policy_1_snips': [1]}, estimate_bucket(df, config))
+
+    def  test_estimate_bucket_empty(self):
+        df = {}
+        config = {}
+        self.assertEqual({}, estimate_bucket({},{}))
+
+    def test_estimate_bucket_only_ips(self): # need to use estimate_bucket
+        df = pd.DataFrame({'p': [0.5], 'r': [1], 'policy_1': [0.5]})
```

Comment:
Just in case if you don't how to mapping the df data with the `add_example` api in estimator lib. `'p[0]'` is `'p_log'`, `'policy_1[0]'` is `'p_pred'`